java/arcs/jvm/host/AnnotationBasedJvmProdHost.kt

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2019 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+package arcs.jvm.host
+
+import arcs.core.host.ArcHost
+import arcs.core.host.ParticleIdentifier
+import arcs.core.host.ParticleRegistration
+import arcs.core.host.ProdHost
+import arcs.core.host.api.Particle
+import arcs.core.host.toParticleIdentifier
+import arcs.core.util.Time
+import arcs.jvm.util.JvmTime
+import java.util.ServiceLoader
+import kotlin.reflect.KClass
+
+/**
+ * An [ArcHost] that runs isolatable particles that are expected to have no platform
+ * dependencies. Automatically scans class path using [ServiceLoader] to find additional particles.
+ *
+ * @property targetHost the [TargetHost] class this host will register particles for.
+ * @property additionalParticles extra [Particle]s not automatically scanned from classpath.
+ */
+open class AnnotationBasedJvmProdHost(
```

Comment:
Does this need to be a subclass? Couldn't any of the `Host` classes use this approach?

Reply:
True, it could just be a utility function.

Reply:
Let's do that in a cleanup.